src/pages/index.js

```diff
@@ -1,103 +1,188 @@
-    link: "https://www.amtrakvacations.com/sites/amtrak/files/styles/hero/public/media/images/grand-canyon_670151752-web.jpg?h=f790d5ec&itok=PmHMpYK5",
-  },
-];
+// VARIABLES
+
+const config = {
+  formSelector: ".modal__form",
+  inputSelector: ".modal__form-input",
+  submitButtonSelector: ".modal__save-button",
+  inactiveButtonClass: "modal__save-button_disabled",
+  inputErrorClass: "modal__error",
+  errorClass: "modal__form-input_type_error",
+};
 
 // Profile Variables
 
-const profileEditModal = document.querySelector("#profile-edit-modal");
-const profileTitle = document.querySelector(".profile__title");
-const profileSubtitle = document.querySelector(".profile__subtitle");
 const profileTitleInput = document.querySelector("#profile-title-input");
 const profileSubtitleInput = document.querySelector("#profile-subtitle-input");
 const profileForm = document.forms["modal-form"];
+const editImageForm = document.querySelector("#image-edit-form");
+const avatarImage = document.querySelector("#profile-avatar-image");
+const profileEditImageForm = document.querySelector("#image-edit-form");
+const pencil = document.querySelector("#profile-pencil");
 
 // Card Variables
 
 const addCardModal = document.querySelector("#add-card-modal");
 const addCardForm = addCardModal.querySelector("#add-card-form");
-const cardTitleInput = document.querySelector("#card-title-input");
-const cardUrlInput = document.querySelector("#modal-url-input");
 const cardsWrap = document.querySelector(".cards__list");
 
 // Preview Image Variables
 
 const previewModal = document.querySelector("#preview-modal");
 const modalImage = previewModal.querySelector(".modal__image");
-const previewCaption = previewModal.querySelector(".modal__caption");
 
 // Buttons
 
 const profileEditBtn = document.querySelector("#profile-edit-button");
 const addNewCardBtn = document.querySelector("#add-card-button");
-const closeBtns = document.querySelectorAll(".modal__close-button");
 
-// FUNCTIONS
+// API'S
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "250daeea-0b63-48f9-a4b3-e529116433c4",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    newCardSection.setItems(cards);
+    newCardSection.renderItems();
+  })
+  .catch((err) => {
+    alert(`${err} Failed to get cards.`);
+  });
+
+api
+  .getUserInfo()
+  .then((info) => {
+    profileUserInfo.setUserInfo({
+      name: info.name,
+      description: info.about,
+    });
+    profileUserInfo.setUserAvatar(info.avatar);
+  })
+  .catch((err) => {
+    alert(`${err} Failed to get user info.`);
+  });
+
+// API FUNCTIONS
 
 function handleProfileEditSubmit(data) {
-  profileUserInfo.setUserInfo(data);
+  profileEditForm.setLoading(true);
+  api
+    .updateUserInfo(data)
+    .then((result) => {
+      profileUserInfo.setUserInfo(data);
+      profileEditForm.setLoading(false);
```

Comment:
Needs correction: Change button state in `finally` block, otherwise the button text won’t change back if an error occurred (fix that in other places too)

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally